addons/website_sale_stock/models/product_template.py

```diff
@@ -1,20 +1,34 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models, api
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
 
 
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
-    inventory_availability = fields.Selection([
-        ('never', 'Sell regardless of inventory'),
-        ('always', 'Show inventory on website and prevent sales if not enough stock'),
-        ('threshold', 'Show inventory below a threshold and prevent sales if not enough stock'),
-        ('custom', 'Show product-specific notifications'),
-    ], string='Inventory Availability', help='Adds an inventory availability status on the web product page.', default='never')
+    def _get_default_message(self, key):
+        default_message = self.env.ref('website_sale_stock.default_' + key, raise_if_not_found=False)
+        return default_message._render() if default_message else ''
+
+    allow_to_order = fields.Selection([
+        ('available', 'Allow the user to order only if there is enough quantity on hand for the product'),
+        ('always', 'Allow the user to order even if there is no quantity on hand for the product'),
+    ], string='Allow to order', default='always')
     available_threshold = fields.Float(string='Availability Threshold', default=5.0)
-    custom_message = fields.Text(string='Custom Message', default='', translate=True)
+    available_message = fields.Html(string="In Stock", default=lambda self: self._get_default_message('available_message'))
```

Comment:
All those fields message on product can be droped, the message is not supposed to be overide by product, they all share the same message that you edit through settings
It is only the mode (always/never/..) that should be customizable per product